templates/azure-health.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# Based on https://learn.microsoft.com/en-us/azure/virtual-machines/linux/no-agent#add-required-code-to-the-vm
+
+
+attempts=1
+until [ "$attempts" -gt 2 ]
+do
+    echo "obtaining goal state - attempt $attempts"
+    exec 3<>/dev/tcp/168.63.129.16/80
+    echo -e "GET /machine/?comp=goalstate HTTP/1.1\r
+Host: 168.63.129.16\r
+User-Agent: curl/7.86.0\r
+Accept: */*\r
+x-ms-agent-name: azure-vm-register\r
+Content-Type: text/xml;charset=utf-8\r
+x-ms-version: 2012-11-30\r
+Connection: close\r
+\r" >&3
+    goalstate=$( timeout 1 cat <&3 )
+    
+    if [ $? -eq 0 ]
+    then
```

Comment:
```suggestion
    if [ $? -eq 0 ]; then
```

Again perhaps we could use the nicer `(())` syntax? What u think?